File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        if(!head) {
            return head;
        }
        ListNode* curr = head;
        ListNode* next = head->next;
        
        while(next && curr) {
            int t = curr->val;
            curr->val = next->val;
            next->val = t;
            
            curr = next->next;
            if(curr)
                next = curr->next;
        }
        
        return head;
    }
};

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swap(a *ListNode, b *ListNode) {
    if b == nil {
        return
    }
    
    t := a.Val
    a.Val = b.Val
    b.Val = t
}

func swapPairs(head *ListNode) *ListNode {
    if head == nil || head.Next == nil {
        return head
    }
    var ptr *ListNode
    ptr = head
    
    for ptr != nil && ptr.Next != nil {
        swap(ptr, ptr.Next)
        ptr = ptr.Next.Next
    }
    
    return head
}